### PR TITLE
Add dynamic-configuration-example app

### PR DESCRIPTION
**Pull request by alexandern17**

### CODE REVIEW

File: examples/dynamic-configuration-example/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+This example app demonstrates how to dynamically configure the Thunderhead SDK.
+
+App Flow Summary:
+1. On initial app start, [MainActivity.kt](https://github.com/thunderheadone/one-sdk-android/blob/master/examples/dynamic-configuration-example/app/src/main/java/com/thunderhead/dynamicconfigurationexample/MainActivity.kt) is presented with no region selected/configured.
+2. The 'SELECT' button presents [ChangeRegionActivity.kt](https://github.com/thunderheadone/one-sdk-android/blob/master/examples/dynamic-configuration-example/app/src/main/java/com/thunderhead/dynamicconfigurationexample/ChangeRegionActivity.kt), which displays a list of regions to select from.
+3. Upon selection of a new region, the SDK is *reconfigured* with new API credentials corresponding to the region selected, handled in the [`onActivityResult`](https://github.com/thunderheadone/one-sdk-android/blob/master/examples/dynamic-configuration-example/app/src/main/java/com/thunderhead/dynamicconfigurationexample/MainActivity.kt#L57) method.
+
+## SDK initialization not required
+
+The Thunderhead SDK is automatically initialized in an `un-configured` state.
```
Comment (andreipop) [inline examples/dynamic-configuration-example/README.md:14]: unconfigured

File: examples/dynamic-configuration-example/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+* While `un-configured`, the SDK will continue to locally queue end-user data and will upload the data to server once the SDK is configured with valid parameters.
```
Comment (andreipop) [inline examples/dynamic-configuration-example/README.md:15]: unconfigured

File: examples/dynamic-configuration-example/app/src/main/java/com/thunderhead/dynamicconfigurationexample/DynamicConfigurationApplication.kt [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+         *
+         * The Thunderhead SDK is automatically initialized in an `un-configured` state.
+         *  While `un-configured`, the SDK will continue to locally queue end-user data and will upload the data to server once
+         *  the SDK is configured with valid parameters.
+         *  This can be disabled at any time by setting the `oneOptOutConfiguration` to `true`.
+         *
+         * The SDK does not support partial, or piecemeal, configuration. All parameters must be provided, either all valid or invalid (`empty string` or `null`).
+         * When configured with invalid parameters, it will set the SDK into an `un-configured` state.
+         *
+         * https://github.com/thunderheadone/one-sdk-android/tree/master/examples/dynamic-configuration-example
```
Comment (JustinTullgren) [inline examples/dynamic-configuration-example/app/src/main/java/com/thunderhead/dynamicconfigurationexample/DynamicConfigurationApplication.kt:80]: It links to itself? please remove

File: examples/dynamic-configuration-example/README.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+![Nevercode build status](https://app.nevercode.io/api/projects/a1c694b2-2e86-4a3b-9d0e-9671c5819b9f/workflows/d7a9ef3c-a383-4f1e-a4b1-2c6ed83ce9e7/status_badge.svg?branch=master&style=shields)
```
[the remark this answers is not in the document]
Comment (alexandern17) [reply examples/dynamic-configuration-example/README.md:1]: Yes this is up to date.

**Pull request merged**